File: app/src/main/java/com/example/catchy/fragments/BioDialogFragment.java
package com.example.catchy.fragments;

import android.graphics.Color;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;
import androidx.fragment.app.Fragment;
import androidx.palette.graphics.Palette;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.example.catchy.R;
import com.example.catchy.models.User;
import com.parse.ParseUser;
import com.squareup.picasso.Picasso;

import co.revely.gradient.RevelyGradient;


public class BioDialogFragment extends DialogFragment {
    private TextView tvFullName;
    private TextView tvBio;

    private String fullName;
    private String bio;

    private RelativeLayout layout;

    public BioDialogFragment() {
        // Required empty public constructor
    }

    public static BioDialogFragment newInstance(String fullName, String bio) {
        BioDialogFragment fragment = new BioDialogFragment();
        Bundle args = new Bundle();
        args.putString("fullName", fullName);
        args.putString("bio", bio);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            this.fullName = getArguments().getString("fullName");
            this.bio = getArguments().getString("bio");
        }
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_bio_dialog, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        tvFullName = view.findViewById(R.id.tvFullName);
        tvBio = view.findViewById(R.id.tvBio);
        layout = view.findViewById(R.id.biodialog);

        tvFullName.setText(fullName);
        tvBio.setText(bio);
        super.onViewCreated(view, savedInstanceState);

        setBackgroundColor();
    }

    private void setBackgroundColor() {
        if (User.profileBitmap != null && !User.profileBitmap.isRecycled()) {
            Palette palette = Palette.from(User.profileBitmap).generate();
            Palette.Swatch swatch = palette.getDarkVibrantSwatch();
            // int color = palette.getDarkMutedColor(0);
            if (swatch == null) {
                swatch = palette.getDominantSwatch();
            }

            // swatch.getRgb()
            if (swatch != null) {

                int color = swatch.getRgb();
                RevelyGradient
                        .linear()
                        .colors(new int[]{Color.parseColor("#000000"), color}).angle(270f).alpha(0.66f)
                        .onBackgroundOf(layout);
            }
        }
        else { // try again
            new Thread(() -> {
                try {
                    User.profileBitmap = Picasso.get().load(ParseUser.getCurrentUser().getParseFile("profilePic").getUrl()).get();
                } catch (Exception e) {
                    Log.e("BioDialogFragment", "couldn't get bitmap"+e);
                }
            }).start();
            RevelyGradient
                    .linear()
                    .colors(new int[]{Color.parseColor("#000000"), Color.parseColor("#00EDFF")}).angle(270f).alpha(0.66f)
                    .onBackgroundOf(layout);
        }
    }
}

File: app/src/main/java/com/example/catchy/fragments/BioDialogFragment.kt
package com.example.catchy.fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.RelativeLayout
import android.widget.TextView
import androidx.fragment.app.DialogFragment
import com.bumptech.glide.Glide
import com.example.catchy.R
import com.example.catchy.databinding.FragmentBioDialogBinding

class BioDialogFragment : DialogFragment() {
    private var fullName: String? = null
    private var bio: String? = null
    private var imgUrl: String? = null
    private var layout: RelativeLayout? = null
    private var _binding: FragmentBioDialogBinding? = null
    private val binding get() = _binding!!

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (arguments != null) {
            fullName = requireArguments().getString("fullName")
            bio = requireArguments().getString("bio")
            imgUrl = requireArguments().getString("imgUrl")
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        _binding = FragmentBioDialogBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        val tvFullName : TextView = binding.tvFullName
        val tvBio : TextView = binding.tvBio
        layout = binding.biodialog
        val ivProfileImage : ImageView = binding.ivProfileImage
        tvFullName.text = fullName
        tvBio.text = bio
        Glide.with(this).load(imgUrl).into(ivProfileImage)
        super.onViewCreated(view, savedInstanceState)
    }

    companion object {
        @JvmStatic
        fun newInstance(fullName: String?, bio: String?, imgUrl: String?): BioDialogFragment {
            val fragment = BioDialogFragment()
            val args = Bundle()
            args.putString("fullName", fullName)
            args.putString("bio", bio)
            args.putString("imgUrl", imgUrl)
            fragment.arguments = args
            return fragment
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
}